src/GameManager.cpp

```diff
@@ -316,10 +279,102 @@ unsigned int GameManager::createBarricade(SDL_Renderer* gRenderer, float x, floa
 }
 
 
-void GameManager::deleteBarricade(unsigned int id) {
-    barricadeManager.erase(id);
+void GameManager::deleteBarricade(const int32_t id) {
+ barricadeManager.erase(id);
 }
 // Get a barricade by its id
-Barricade& GameManager::getBarricade(unsigned int id) {
+Barricade& GameManager::getBarricade(const int32_t id) {
     return barricadeManager.find(id)->second;
 }
+
+// Create zombie add it to manager, returns success
+int32_t GameManager::createWall(SDL_Renderer* gRenderer, const float x, const float y, const int w, const int h) {
+    const int32_t id = generateID();
+    objectManager[id] = Wall(w, h);
+    printf("%d\n", id);
+    if (!objectManager.at(id).texture.loadFromFile("assets/texture/wall.png", gRenderer)) {
+        printf("Failed to load the wall texture!\n");
+        deleteBarricade(id);
+        return -1;
+   } else {
+        objectManager.at(id).texture.setDimensions(w, h);
+    }
+    
+    objectManager.at(id).setPosition(x,y);
+    
+    return id;
+}
+
+
+void GameManager::setBoundary(SDL_Renderer* gRenderer, const float startX, const float startY, const float endX, const float endY){
+
+    int width = endX - startX + 200;
+    int height = 100;
+
+    const float x = startX - 100;
+    const float y = startY - 100;
+
+    createWall(gRenderer, x, y, width, height);
+    createWall(gRenderer, x, endY, width, height);
+ 
+    width = 100;
```

Comment:
Don't reassign variable when it isn't clear.
Use separate variable for different walls.